src/axom/core/ArrayBase.hpp

```diff
@@ -18,6 +18,9 @@
 // C/C++ includes
 #include <iostream>  // for std::cerr and std::ostream
 #include <numeric>   // for std::accumulate
+#if defined(__GLIBCXX__) && !defined(_GLIBCXX_USE_CXX11_ABI)
```

Comment:
We only use this fallback for compilers using GCC 4.9.3's standard library, since the C++11 type traits we needed weren't implemented until GCC 5. The TR1 fallback should be available for that compiler library version; for other libraries, we'd just expect full C++11 compliance.